module-system/sov-modules-macros/src/dispatch/default.rs

```diff
@@ -0,0 +1,62 @@
+use syn::DeriveInput;
+
+use super::common::{StructFieldExtractor, StructNamedField};
+
+pub(crate) struct DefaultMacro {
+    field_extractor: StructFieldExtractor,
+}
+
+impl DefaultMacro {
+    pub(crate) fn new(name: &'static str) -> Self {
+        Self {
+            field_extractor: StructFieldExtractor::new(name),
+        }
+    }
+
+    pub(crate) fn derive_default(
+        &self,
+        input: DeriveInput,
+    ) -> Result<proc_macro::TokenStream, syn::Error> {
+        let DeriveInput {
+            data,
+            ident,
+            generics,
+            ..
+        } = input;
+
+        let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
+
+        let fields = self.field_extractor.get_fields_from_struct(&data)?;
+        let runtime_fn_body = Self::make_default_fn_body(&fields);
+
+        // Implements the Default Config trait
+        Ok(quote::quote! {
+        impl #impl_generics Default for #ident #type_generics #where_clause {
```

Comment:
We need a fully path here `std::default::Default`.

Reply:
I belive it should be even `::std...`

Reply:
I caught up with @citizen-stig. This is a good suggestion. I was not aware of this trick in macros before.